WordPress/WordPressShareExtension/ShareViewController.swift

```diff
@@ -243,66 +277,165 @@ private extension ShareViewController {
         textView.text = content
     }
 
-    func imageLoaded(image: UIImage) {
-        // Load the View
+    func imagesLoaded(images: [UIImage]) {
         let mediaView = MediaView()
-        mediaView.resizeIfNeededAndDisplay(image)
+        if let image = images.first {
+            // Load the View with the first image only
+            mediaView.resizeIfNeededAndDisplay(image)
+        }
 
         // References please
-        self.mediaImage = image
+        self.mediaImages = images
         self.mediaView = mediaView
         self.reloadConfigurationItems()
     }
 }
 
-/// ShareViewController Extension: Backend Interaction
-///
+// MARK: - ShareViewController Extension: Persistence
+
+private extension ShareViewController {
```

Comment:
If it's ok with you @frosty , I think I am going to move this logic out of the VC since we have a new one replacing it very soon (and it could be shared with WPiOS as well).